lib/ansible/utils/display.py

```diff
@@ -254,54 +249,68 @@ def verbose(self, msg, host=None, caplevel=2):
             else:
                 self.display("<%s> %s" % (host, msg), color=C.COLOR_VERBOSE, stderr=to_stderr)
 
-    def deprecated(self, msg, version=None, removed=False, date=None):
+    def get_deprecation_message(self, msg, version=None, removed=False, date=None, collection_name=None):
         ''' used to print out a deprecation message.'''
+        msg = msg.strip()
+        if msg[-1] not in ['!', '?', '.']:
```

Comment:
Maybe
```suggestion
        if msg and msg[-1] not in ['!', '?', '.']:
```
to avoid crashes if `msg` is empty.